[fix] コードが折り返されない不具合を修正

TatsuruHisanaga opened the pull request:
## 変更の概要
`hilight.js`によるコードのシンタックスハイライトが折り返されず、１行のコードが長い場合に意図した要素からはみ出されてしまう不具合を修正しました。
#8 #9

変更後の画像：
![image](https://github.com/fixmix-hackathon/fixmix_frontend/assets/119847001/dff4884c-eac0-4cba-a462-2b3fa090080d)

> この変更により、コードの折り返しが有効になりました。（画像の青い選択部分）

## なぜこの変更をするのか
コードが折り返されなかった場合、コンテンツ全体の横幅が画面幅よりも広くなってしまうため、該当するメッセージを消去しない限りはUIが永続的に崩れてしまう問題がありました。

## やったこと

-  新たに`CodeBlock.module.css`ファイルを作成し、デフォルトでコードの折り返しが適用されるようにスタイルを追加した。
- うまく適用できていなかったスタイルを消去した。
- famの画像幅を`auto`にし、画面幅が小さくなっても優先的に表示され続けるようにした。
- `marked`ライブラリのオプション`langPrefix`を追加した。

`langPrefix` は、`marked` ライブラリのオプションの一つで、生成されるコードブロックの`<code>`タグに適用されるCSSクラスのプレフィックスを定義します。

例えば、Markdownファイルに以下のようなコードブロックがあるとします：

```markdown
```javascript
console.log('Hello, World!');
```


`marked` がこれをHTMLに変換するとき、`langPrefix` オプションが `'language-'` に設定されていると、生成されるHTMLは以下のようになります：

```html
<pre>
  <code class="language-javascript">
    console.log('Hello, World!');
  </code>
</pre>
```

この `language-javascript` クラスは、シンタックスハイライトライブラリ（例：Prism.jsやHighlight.js）がJavaScriptコードとしてこのブロックをハイライトするために使用します。

このプレフィックスを変更すると、生成されるHTMLのクラス名も変わります。これは、特定のCSSスタイルを適用する場合や、異なるシンタックスハイライトライブラリを使用する場合に便利です。

今回の変更では`langPrefix`の追加はUIに影響しませんが、今後のことを考えて導入してみました。

TatsuruHisanaga committed:
[fix] #9 famのアバター画像が隠れてしまう不具合を修正

## 詳細
現状では画面サイズが440pxを下回るとfamのアバター画像が左にずれて隠れるようになってしまっていたが、今回の修正によってfamの画像が如何なる画面幅でも問題なく表示されるようにした。

## 課題
現状ではfamの初期メッセージの影響なのか、画面幅が360px未満の端末では依然としてfamの画像が隠れてしまう不具合が確認された。（iPhone 5やiPhoneSE 2016）

## 今後の対応
hilight.js でコードの折り返しが有効になるようにする。

TatsuruHisanaga committed:
[fix] コードブロックで折り返しが効かない不具合を修正

## 原因
markedとhighlight.jsのスタイリングがうまく適用されていなかった。

## 対応
CSS modulesファイルを作成し、新たにスタイルを追加、更新した。

この修正により、コードがうまく折り返されるようになった他、画面幅が小さくなると一部の要素が画面からはみ出てしまう不具合も解消した。

TatsuruHisanaga committed:
Merge branch 'fix/responsive_style_bug' into develop

KO1231 commented on app/components/CodeBlock.module.css:
```diff
@@ -0,0 +1,4 @@
+.markdown pre, .markdown code {
```
### **master**
![image](https://github.com/fixmix-hackathon/fixmix_frontend/assets/124903312/22d5eb71-ff22-4977-8e2c-cb6c8496b0e3)

### **develop**
![image](https://github.com/fixmix-hackathon/fixmix_frontend/assets/124903312/d8a456eb-c82d-4f3a-9328-58f6d2be2104)

と1行あたり長いコードを無理やり出力させたとき、折り返されるのを確認できませんでした。

KO1231 commented on app/components/Chat.tsx:
```diff
@@ -74,7 +76,7 @@ const Chat = ({ role, content, fromStorage }: Message) => {
             >
                 <Avatar
                     name={role === "user" ? "Me" : "Fam"}
-                    w="45px"
```
### **master**
![image](https://github.com/fixmix-hackathon/fixmix_frontend/assets/124903312/c054cc09-c9c5-45bb-bcec-8ff3ffe1da94)

### **develop**
![image](https://github.com/fixmix-hackathon/fixmix_frontend/assets/124903312/8cc7c112-26f0-4db9-97f3-e35afd795d46)

と手持ちのiOS環境でfαmの描画が切れている状態になっています。
環境としては、iPhone 14 Plus (iOS 16.6, Safari 16.6) 、iPhone 7 Plus(iOS 15.7.5,  Safari 15.6.4)です。
なお、Android環境、Windows環境(ともにGoogle Chrome)では特に描画切れ等確認できなかったので、Safari特有等一部環境で見られるものの可能性があります。

KO1231 commented on app/components/Chat.tsx:
```diff
@@ -132,9 +134,7 @@ const Chat = ({ role, content, fromStorage }: Message) => {
                         dangerouslySetInnerHTML={{ __html: marked.parse(chatMessage) }}
-                        style={{
-                            whiteSpace: 'pre-wrap', // overflowWrap: 'break-word'
-                        }}
```
### **master**
![image](https://github.com/fixmix-hackathon/fixmix_frontend/assets/124903312/c054cc09-c9c5-45bb-bcec-8ff3ffe1da94)

### **develop**
![image](https://github.com/fixmix-hackathon/fixmix_frontend/assets/124903312/8cc7c112-26f0-4db9-97f3-e35afd795d46)

whiteSpace: 'pre-wrap' が適用されていないようです。class化したときの適用がうまくいっていない可能性があります。

KO1231 reviewed (changes requested):
コードの折返しが確認できませんでした。
また、一部環境(reviewer確認でiOS)においてfαmアイコンの描画に不具合と、出力文章における改行の一部消失を確認しました。
（詳細はコメント）

TatsuruHisanaga committed:
[change] #8 `overflow-wrap` プロパティを `anywhere` に変更

## 詳細
このコミットでは、CSSプロパティを `word-wrap: break-word !important;` から `overflow-wrap: anywhere !important;` に変更しました。

`word-wrap: break-word` の使用時には、長い単語やURLが任意の場所で分割されることがありますが、この分割位置がコンテンツの最小固有寸法（コンテンツが占める最小の幅）の計算時には考慮されません。これが原因で、レイアウトの問題やテキストのオーバーフローが発生する可能性がありました。

新しい `overflow-wrap: anywhere` プロパティを採用することで、ブラウザは長い単語やURLを任意の場所で分割し、新しい行に移動させることが可能となります。さらに、この新しい分割位置はコンテンツの最小固有寸法を計算する際に考慮されるため、テキストがコンテナからはみ出さないように調整されます。これにより、テキストのオーバーフローを避け、ウェブページの読みやすさとレイアウトの安定性が向上します。

※ "overflow-wrap"は"word-wrap"の新しい名称です。

TatsuruHisanaga committed:
Merge branch 'fix/responsive_style_bug' into develop

TatsuruHisanaga committed:
[add] userのメッセージに行の折り返しを適用

TatsuruHisanaga committed:
[add] codeタグにスタイルを追加

TatsuruHisanaga committed:
[delete] #10  ヘッダーを非表示にする処理の消去

TatsuruHisanaga committed:
[improve] コードの整形

## 更新内容
1. セミコロンを統一
2. シングルクオートを統一
3. インデントを統一

TatsuruHisanaga committed:
[clean] CodeBlock.module.css -> Markdown.module.css

## 変更内容
CSS Modulesファイルの名称を'CodeBlock.module.css'から'Markdown.module.css'に変更しました。
## 理由
このファイルの中身で行われているスタイリングはコードブロックのみならず、マークダウン形式全体に適用されるものであるため。

TatsuruHisanaga commented:
@KO1231 
一応修正できそうなんやけども、これって別のプルリク出したらまずいんかな😅
このプルリク内で新しくマージのレビューとかできるん？
